src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,286 @@
+        b_list, intr_list, 
+        a_idx_list, b_idx_list, 
+        alpha_b_list
+    )::b_neighbors, b_list
+
+    This function builds b_list. Note that after calling this function, b_list
+    is not fully updated. The entry/exit flags still need to be updated. The variables poly_a and poly_b are the original polygons that the
+    user passed in. The variables edges_a and edges_b are lists of the edges of each polygon, 
+    in the form of their cartesian points. Then intr_list is a list of the intersection points
+    in their cartesion point form. Then a_idx_list is a list of the indicies of where in a_list
+    an intersection point lies. The value at index i of a_idx_list is the location in a_list where
+    the ith point of intr_list lies. Then b_idx_list indicates which EDGE of poylgon b an intersection
+    point lies. Then alpha_b_list is a veector of scalar values which help order
+    the intersection points in the case that they lie on the same edge of a polygon. Finally,
+    b_list starts out as an array of PolyNodes only containing the original points of poly_b
+    but after this function is run it include intersection points to.
+"""
+
+function _build_b_list(b_list, intr_list, a_idx_list, b_idx_list, alpha_b_list)
+    # Iterate through the b_list and add in intersection points
+    # Occasionally I need to skip the new points I added to the array
+    skip = false
+    num_skips = 0
+    b_neighbors = Array{Int, 1}(undef, length(intr_list))
+    for ii in 1:(length(b_list)+length(intr_list))
+        if skip
+            num_skips = num_skips - 1
+            if num_skips == 0
+                skip = false
+            end
+            continue
+        end
+        i = findall(x->x==b_list[ii].idx, b_idx_list)
+        if !isempty(i)     
+            # Order intersection points based on alpha values
+            new_order = sortperm(alpha_b_list[i])
+            pts_to_add = Array{PolyNode, 1}(undef, length(i))
+            for m in eachindex(i)
+                pts_to_add[new_order[m]] = PolyNode(i[m], true, a_idx_list[i[m]], false, alpha_b_list[i[m]])
+                b_neighbors[i[m]] = ii + new_order[m]
+            end   
+            # I use splice instead of insert so I can insert array   
+            splice!(b_list, ii+1:ii, pts_to_add)
+            skip = true
+            num_skips = length(i)
+        end
+    end
+
+    return b_neighbors, b_list
+end
+
+"""
+    _flag_ent_exit(
+        ::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, 
+        poly_b, a_list, b_list, edges_a, edges_b
+    )::a_list, b_list
+
+    This function flags all the intersection points as either an 'entry' or 'exit' point.
+"""
+function _flag_ent_exit(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b, a_list, b_list, edges_a, edges_b)
+    # Put in ent exit flags for poly_a
+    status = false
+    for ii in eachindex(a_list)
+        if ii == 1
+            temp = within(edges_a[ii][1], poly_b)
+            status = !(temp[1])
```

Comment:
`within` should just return a boolean, so don't need the indexing. 

Also this should just be a helper function (lines 186 - 199) that takes in a `poly` and a `poly_list` and then you call it twice, in your main function for both `poly_a` and `poly_b` since the code is repeated. 

Also, don't need `edges_a` and `edges_b`. And since it is helper function that we will pass stuff to, you don't need to check that `poly_a` and `poly_b` are polygons.

Reply:
fixed!